pkgs/development/compilers/llvm/7/bintools.nix

```diff
@@ -0,0 +1,19 @@
+{ runCommand, stdenv, llvm, lld, version }:
+
+let
+  prefix =
+    if stdenv.hostPlatform != stdenv.targetPlatform
+    then "${stdenv.targetPlatform.config}-"
+    else "";
+in runCommand "llvm-binutils-${version}" { preferLocalBuild = true; } ''
+   mkdir -p $out/bin
+   for prog in ${lld}/bin/*; do
+     ln -s $prog $out/bin/${prefix}$(basename $prog)
+   done
+   for prog in ${llvm}/bin/*; do
+     ln -s $prog $out/bin/${prefix}$(echo $(basename $prog) | sed -e "s|llvm-||")
+     ln -sf $prog $out/bin/${prefix}$(basename $prog)
+   done
+   rm -f $out/bin/${prefix}cat
+   ln -s ${lld}/bin/lld $out/bin/${prefix}ld
```

Comment:
LOL I see my comment re:using llvm bits symlinked over ... is not news to you! :dagger: 

FWIW this might be a great deal simpler and more predictable if instead of shell globs and loops it used `linkFarm` or whichever over in `trivial-builders.nix`.  I think enumerating what we intend/expect to symlink is reasonable- the number shouldn't grow much higher than it is and this way nothing sneaks in as part of a bump :).

Reply:
Yeah that probably would be better. Even just listing out all of the utilities here like we do in https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/binutils/default.nix would be helpful. This is actually originally from wasm-cross here:

https://github.com/WebGHC/wasm-cross/blob/master/llvm-head/default.nix#L58-L82

but I think that change would be a good idea.